### PR TITLE
feat: 🎸 adjust existing fetch map and user endpoints to fetch based on status

### LINKED ISSUE #1785: Adjust existing fetch map(s) and user(s) endpoints to fetch based on status
Existing endpoints for fetching maps and users should be adjusted to fetch based on the map status (for the former) or user status (for the latter).

### PR DESCRIPTION
**Pull request by PeculiarE**
Fixes #1785

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,6 +2,9 @@ class ApplicationController < ActionController::Base
   # include OpenIdAuthentication # shouldn't be necessary!!
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
+
+  include ActionView::Helpers::DateHelper # required for time_ago_in_words()
```
Comment (jywarren) [inline app/controllers/application_controller.rb:6]: Nice and i appreciate the comment!

**Review by jywarren — CHANGES_REQUESTED** [review]
Hi @PeculiarE I'm sorry if I missed this but i left a comment about the "moderated by default" and its impact on moderator duties. But I think an adjustment to the criteria for publishing as I suggested would be not a huge change to this PR and would work. It would still reduce spam but wouldn't place a burden on moderators to approve each user.

**Comment by jywarren** [comment]
Great work, this PR was super readable and a great size for reviewing!

**Comment by jywarren** [comment]
Well, upon creating a new map, if it's moderated by default, then all new
authors will require a moderator to sign off. If instead we just allow
moderators to go in and remove spam maps, then forbid those users from
making future maps, we don't add a lot of extra work for the moderators,
although some more spam will come through before they catch it.

**Comment by PeculiarE** [comment]
So, in essence, no moderation of new maps? All new maps should be published immediately?

**Comment by jywarren** [comment]
Yes, that's right. Moderation will be upon review only. Well, we can ask
the community reps to confirm this, what do you think?

**Comment by jywarren** [comment]
We can mention @ebarry and @publiclab/community-reps as well as reach out
over the chatroom - if to Liz, it can be in the public channel as well.
Thanks!

**Comment by PeculiarE** [comment]
Alright. Thank you!

**Comment by jywarren** [comment]
@ebarry this applies to anonymous maps as well, right?

@PeculiarE do you think there are parts of this PR we should preserve for other variations of the rules?

I wanted to say I should have caught this as a policy change earlier and I'm sorry for the lost time you spent crafting it @PeculiarE -- I appreciate your efforts. Thank you!

**Comment by PeculiarE** [comment]
Thank you 😊 @jywarren....I did learn something new in Rails and also realised my misunderstanding/mixup of the first-time-author and new-contributor concepts while working on this. So, there's a silver lining 😄 

> @PeculiarE do you think there are parts of this PR we should preserve for other variations of the rules?

Hmmm....I really can't say which parts are worth preserving at this point. I think it depends on knowing what these variations are/will be. Then we can decide which parts to keep.

I removed the entire commit a few hours ago though 🤦‍♀️ ....but I could easily rewrite the parts we decide that we need

**Comment by PeculiarE** [comment]
Here's a preview of the new behavior when fetching a single map or a single user if the map has been spammed and/or author has been banned.

Admins/mods can see the status details while other users are redirected.

https://user-images.githubusercontent.com/75625011/180181946-e2b72cfc-313a-4cbb-a775-ee3336472a01.mp4

Is this okay @jywarren, @cesswairimu, @TildaDares?

**Comment by jywarren** [comment]
Yes, this looks super! Is this ready to merge then? Great work!!

**Comment by PeculiarE** [comment]
Thank you @jywarren, and yeah, it's okay to merge.

**Review by cesswairimu — APPROVED** [review]
looks great :rocket:

**Comment by jywarren** [comment]
🎉

**Pull request merged**